controllers/apim/virtualservice_controller.go

```diff
@@ -0,0 +1,216 @@
+package apim
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	networkingv1alpha3 "istio.io/api/networking/v1alpha3"
+	securityv1beta1 "istio.io/api/security/v1beta1"
+	istionetworkingv1alpha3 "istio.io/client-go/pkg/apis/networking/v1alpha3"
+	istiosecurityv1beta1 "istio.io/client-go/pkg/apis/security/v1beta1"
+
+	"github.com/go-logr/logr"
+	"github.com/kuadrant/kuadrant-controller/pkg/log"
+	"github.com/kuadrant/kuadrant-controller/pkg/reconcilers"
+)
+
+const (
+	KuadrantAuthProviderAnnotation = "kuadrant.io/auth-provider"
+)
+
+// VirtualServiceReconciler reconciles Istio's AuthorizationPolicy object
+type VirtualServiceReconciler struct {
+	*reconcilers.BaseReconciler
+	Scheme *runtime.Scheme
+}
+
+func (r *VirtualServiceReconciler) Reconcile(eventCtx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	logger := r.Logger().WithValues("VirtualService", req.NamespacedName)
+	ctx := logr.NewContext(eventCtx, logger)
+
+	virtualService := istionetworkingv1alpha3.VirtualService{}
+	if err := r.Client().Get(ctx, req.NamespacedName, &virtualService); err != nil {
+		if apierrors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+		logger.Error(err, "failed to get VirtualService")
+		return ctrl.Result{}, err
+	}
+
+	if err := r.reconcileAuthPolicy(ctx, &virtualService); err != nil {
+		logger.Error(err, "failed to reconcile AuthorizationPolicy")
+		return ctrl.Result{}, err
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func getAuthPolicyName(vsName string) string {
+	return "source-virtualservice-" + vsName
+}
+
+func (r *VirtualServiceReconciler) reconcileAuthPolicy(ctx context.Context, vs *istionetworkingv1alpha3.VirtualService) error {
+	logger := r.Logger()
+	logger.Info("Reconciling AuthorizationPolicy")
+
+	authPolicy := istiosecurityv1beta1.AuthorizationPolicy{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      getAuthPolicyName(vs.Name),
+			Namespace: vs.Namespace,
+		},
+	}
+
+	providerName, present := vs.GetAnnotations()[KuadrantAuthProviderAnnotation]
+	if !present {
+		return fmt.Errorf("Kuadrant auth-provider annotation not found")
```

Comment:
I don't think this needs to be an error? Perhaps we just log and return?

Reply:
Yea, we can just log it. updated.